homeassistant/helpers/area_registry.py

```diff
@@ -76,12 +76,12 @@ class EventAreaRegistryUpdatedData(TypedDict):
 class AreaEntry(NormalizedNameBaseRegistryEntry):
     """Area Registry Entry."""
 
-    aliases: set[str]
+    aliases: frozenset[str]
     floor_id: str | None
     humidity_entity_id: str | None
     icon: str | None
     id: str
-    labels: set[str] = field(default_factory=set)
+    labels: frozenset[str] = field(default_factory=frozenset)
```

Comment:
It would be simpler to allow passing in either a set or frozenset, and automatically convert to a frozenset.
Unfortunately, dataclasses don't have converters and it doesn't look like it's coming anytime soon: https://peps.python.org/pep-0712/

Maybe it would be better to make `AreaEntry` an `attr` class instead?